connaisseur/crypto.py

```diff
@@ -10,11 +12,26 @@ def verify_signature(public_base64: str, signature_base64: str, message: str):
 
     Raises ValidationError if unsuccessful.
     """
-    public = base64.b64decode(public_base64)
-    pub_key = ecdsa.VerifyingKey.from_der(public)
+    pub_key = decode_and_verify_ecdsa_key(public_base64)
 
     signature = base64.b64decode(signature_base64)
 
     msg_bytes = bytearray(message, "utf-8")
 
     return pub_key.verify(signature, msg_bytes, hashfunc=hashlib.sha256)
+
+
+def decode_and_verify_ecdsa_key(public_base64: str):
+    """
+    Verifies that the provided public key in base64 encoding qualifies as a
+    proper ecdsa key and throws if not.
+    """
+    public = base64.b64decode(public_base64)
+    try:
+        pubkey = ecdsa.VerifyingKey.from_der(public)
+    except ecdsa.keys.MalformedPointError as err:
+        raise InvalidPublicKey(
+            f"The public key provided is no valid ecdsa key: {err}."
+        ) from err
```

Comment:
```suggestion
        )
```

> Exception chaining happens automatically when an exception is raised inside an `except` or `finally` section.
Source: https://docs.python.org/3/tutorial/errors.html

Reply:
pylint does not like that: https://github.com/sse-secure-systems/connaisseur/runs/2295833587?check_suite_focus=true

> crypto.py:33:8: W0707: Consider explicitly re-raising using the 'from' keyword (raise-missing-from)
They like explicit better apparently...

Reply:
Think I should tell pylint to ignore it or make it explicit? In general, explicit is better than implicit, but at this point I am not sure whether that rule is good...

Reply:
well actually there is apparently a difference: https://stefan.sofa-rockers.org/2020/10/28/raise-from/
will re-add the `from err`